rust/src/lib/query_apply/route.rs

```diff
@@ -80,36 +84,84 @@ impl MergedRoutes {
                         ),
                     ));
                 }
-            } else if rt.route_type.is_some() {
-                // In nispor, the IPv4 route with route type `Blackhole`,
-                // `Unreachable`, `Prohibit` does not have the route oif
-                // setting.
-                let mut route_type_rt = rt.clone();
-                if !route_type_rt.is_ipv6() {
-                    route_type_rt.next_hop_iface = None;
+            } else {
+                let mut rt2;
+                if rt.route_type.is_some() && !rt.is_ipv6() {
+                    // In nispor, the IPv4 route with route type `Blackhole`,
+                    // `Unreachable`, `Prohibit` does not have the route oif
+                    // setting.
+                    rt2 = rt.clone();
+                    rt2.next_hop_iface = None;
+                    rt = &rt2
                 }
-                if !cur_routes
-                    .as_slice()
-                    .iter()
-                    .any(|cur_rt| route_type_rt.is_match(cur_rt))
-                {
+
+                if !cur_routes.iter().any(|cur_rt| rt.is_match(cur_rt)) {
+                    if is_route_delayed_by_nm(rt, current_ifaces) {
```

Comment:
Instead of iterate the current interfaces for every miss match route, please store the interfaces with auto ip enabled into a `HashSet` to save the repeat search.